modules/functional_expansion_tools/src/series/CompositeSeriesBasisInterface.C

```diff
@@ -0,0 +1,299 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// Module includes
+#include "CompositeSeriesBasisInterface.h"
+#include "Legendre.h"
+
+/*
+ * Default constructor creates a functional basis with one term. In order for the _series member to
+ * be initialized, we initialized it with a Legendre series.
+ */
+CompositeSeriesBasisInterface::CompositeSeriesBasisInterface() : FunctionalBasisInterface(1)
+{
+  _series.push_back(new Legendre());
+}
+
+/*
+ * The non-default constructor is where we actually loop over the series_types and initialize
+ * pointers to those members. Because we won't know the number of terms until the end of the body
+ * of the constructor, we need to call the default FBI constructor.
+ */
+CompositeSeriesBasisInterface::CompositeSeriesBasisInterface(
+    const std::vector<std::size_t> & orders, std::vector<MooseEnum> series_types)
+  : FunctionalBasisInterface(), _series_types(series_types)
+{
+  if (orders.size() != _series_types.size())
+    mooseError("CSBI: Incorrect number of 'orders' specified for 'FunctionSeries'! Check that "
```

Comment:
Good idea, but I'll pass in a reference to the string instead. This class itself doesn't sub/super class any `MooseObject`. At least it will show which object instance using `CompositeSeriesBasisInterface` is causing the problems.